src/lib.rs

```diff
@@ -538,79 +556,94 @@ where
                 );
             }
 
-            base_address = match entry.attr_value(gimli::DW_AT_low_pc) {
-                Ok(Some(gimli::AttributeValue::Addr(addr))) => addr,
-                Err(e) => {
-                    return Err(Error::from(ErrorKind::Gimli(e)))
-                        .chain_err(|| "invalid low_pc attribute")
-                }
-                _ => match entry.attr_value(gimli::DW_AT_entry_pc) {
-                    Ok(Some(gimli::AttributeValue::Addr(addr))) => addr,
-                    Err(e) => {
-                        return Err(Error::from(ErrorKind::Gimli(e)))
-                            .chain_err(|| "invalid entry_pc attribute")
+            let mut attrs = entry.attrs();
+            while let Some(attr) = attrs
```

Comment:
Hmm.. This now walks all the entries in the abbreviation rather than just the first in DFS order, which means that if an attribute is present in more than one, it'll override the others. Is that the right thing to do? Do we expect there to be many uninteresting entries? If so, we're now iterating over all of them whereas before we didn't do any iteration.

Reply:
There shouldn't be duplicate attributes. I don't think it matters if the behaviour has changed for this case.

Before we iterated multiple times up to a specific attribute. Now we iterate once over all attributes. There aren't many uninteresting attributes, so this is a win.